### PR TITLE
chore: AMR-557 - Update data model

### PR DESCRIPTION
**Pull request by shaileshpandey**
# These changes should be only in 1.7 and higher branches

### CODE REVIEW

File: scst-store/images/amr-data-model.jpg
Comment (ohlookadollar) [inline scst-store/images/amr-data-model.jpg:1]: is it just me or is the jpg cut off?

**Review by ohlookadollar — CHANGES_REQUESTED** [review]

File: scst-store/images/amr-data-model.jpg
Comment (shaileshpandey) [reply scst-store/images/amr-data-model.jpg:1]: I added a small image

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 1]
```diff
@@ -12,26 +12,56 @@ The following diagram shows the data models used in the AMR to store artifact in
 
 There are two types of data models in the AMR:
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: Sounds like there are more than two types now?

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 2]
```diff
@@ -12,26 +12,56 @@ The following diagram shows the data models used in the AMR to store artifact in
...
 - `reference`: unique reference to the location. It is automatically set to be the `kube-system` namespace UID by the AMR. This is not configurable by the user.
-- `alias`: unique alias for the location. 
-  By default, it is initialized with the value of the `reference`, but users have the option to configure it to a different value. See [AMR Configuration](configuration.hbs.md).
-- `labels`: labels of the location. Users can add them by using tap values YAML file. See [AMR Configuration](configuration.hbs.md).
\ No newline at end of file
+- `labels`: labels of the location. Users can add them by using tap values YAML file. See [AMR Configuration](configuration.hbs.md).
+
+### <a id='containers'></a> Containers
+
+The `Containers` data model represents the information of a container which is a runtime information about an image. Observer sends this information to the Cloud Event Handler, so each entry represents the state of the container when an event occurred. See [cloud events](cloudevents.hbs.md).
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: `Observer` -> `The Observer` or `An Observer`

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 3]
```diff
@@ -12,26 +12,56 @@ The following diagram shows the data models used in the AMR to store artifact in
...
+The `Images` data model represents the information of an image which acts as a template for containers. Observer sends this information to the Cloud Event Handler to store newly generate image. See [cloud events](cloudevents.hbs.md).
+
+Each `Image` data entry stores information about the associated location it was deployed in, details about the state, and what image was used. See the diagram above for a list of all fields the data model stores.
+You can only associate an `Images` entry with one `Locations` entry. You can point multiple `Images` entries to the same `Locations` entry. 
+
+### <a id='commits'></a> Commits
+
+The `Commits` data model represents the information of a snapshot of the changes that have been made to a project's files. Observer sends this information to the Cloud Event Handler to store new commits. See [cloud events](cloudevents.hbs.md).
+
+Each `Commit` data entry stores information about the associated location it was deployed in, details about the state, and what image was used. See the diagram above for a list of all fields the data model stores.
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: `associated location it was deployed in` -> `associated location in which it was deployed`

Comment (xtreme-debbie-chen) [reply scst-store/amr/data-model-and-concepts.hbs.md]: `what image was used` - is this accurate for the Commits description?

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 4]
```diff
@@ -12,26 +12,56 @@ The following diagram shows the data models used in the AMR to store artifact in
...
+You can only associate an `Commits` entry with one `Locations` entry. You can point multiple `Commits` entries to the same `Locations` entry. 
+
+### <a id='appAcceleratorRuns'></a> AppAcceleratorRuns
+The `AppAcceleratorRuns` data model represents the information of new projects running from Git repositories. An accelerator.yaml file in the repository declares input options for the accelerator. This file also contains instructions for processing the files when you generate a new project. Observer sends this information to the Cloud Event Handler to store `AppAcceleratorRuns`. See [cloud events](cloudevents.hbs.md).
+
+### <a id='appAcceleratorFragments'></a> AppAcceleratorFragments
+The `AppAcceleratorFragments` data model represents the information of a fragment of accelerator app. Observer sends this information to the Cloud Event Handler to store `AppAcceleratorFragments`. See [cloud events](cloudevents.hbs.md).
+
+`AppAcceleratorRuns` can have zero or may `AppAcceleratorFragments`.
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: `can have zero or may` -> `can have zero or many`

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 5]
```diff
@@ -12,26 +12,56 @@ The following diagram shows the data models used in the AMR to store artifact in
...
+
+Each `Commit` data entry stores information about the associated location it was deployed in, details about the state, and what image was used. See the diagram above for a list of all fields the data model stores.
+You can only associate an `Commits` entry with one `Locations` entry. You can point multiple `Commits` entries to the same `Locations` entry. 
+
+### <a id='appAcceleratorRuns'></a> AppAcceleratorRuns
+The `AppAcceleratorRuns` data model represents the information of new projects running from Git repositories. An accelerator.yaml file in the repository declares input options for the accelerator. This file also contains instructions for processing the files when you generate a new project. Observer sends this information to the Cloud Event Handler to store `AppAcceleratorRuns`. See [cloud events](cloudevents.hbs.md).
+
+### <a id='appAcceleratorFragments'></a> AppAcceleratorFragments
+The `AppAcceleratorFragments` data model represents the information of a fragment of accelerator app. Observer sends this information to the Cloud Event Handler to store `AppAcceleratorFragments`. See [cloud events](cloudevents.hbs.md).
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: As someone who knows very little about AppAcceleratorFragments, I have to say that this doesn't describe it much at all. I don't feel like I know what an AppAcceleratorFragment is any better after reading this description. Can anything be done to make this section more descriptive? Or is Fragment a common vocabulary for AppAccelerator such that anybody familiar with the domain would know what it means?

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 6]
```diff
@@ -12,26 +12,56 @@ The following diagram shows the data models used in the AMR to store artifact in
...
+
+`AppAcceleratorRuns` can have zero or may `AppAcceleratorFragments`. 
+
+### <a id='doraMetrics'></a> DoraMetrics
+The `DoraMetrics` data model represents the information of Dora Metric. Observer sends this information to the Cloud Event Handler to store `DoraMetrics`. See [cloud events](cloudevents.hbs.md).
+
+### <a id='doraMetricsPerCorrelationID'></a> DoraMetricsPerCorrelationID
+The `DoraMetricsPerCorrelationID` data model represents the information of Dora Metric for Correlation ID. Observer sends this information to the Cloud Event Handler to store `DoraMetricsPerCorrelationID`. See [cloud events](cloudevents.hbs.md).
+`DoraMetrics` can have zero or may `DoraMetricsPerCorrelationID`.
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: `can have zero or may` -> `can have zero or many`

File: scst-store/amr/data-model-and-concepts.hbs.md (hunk 5 above)
Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md]: I updated the description little bit using this https://docs.vmware.com/en/VMware-Tanzu-Application-Platform/1.3/tap/GUID-application-accelerator-creating-accelerators-creating-accelerators.html#using-accelerator-fragments-4.

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 7]
```diff
@@ -10,28 +10,58 @@ The following diagram shows the data models used in the AMR to store artifact in
 
 ## <a id='amr-data-models'></a> AMR data models
 
-There are two types of data models in the AMR: 
+There are following types of data models in the AMR:
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: What about something like this: `These are the data models stored in the AMR:`

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 8]
```diff
@@ -10,28 +10,58 @@ The following diagram shows the data models used in the AMR to store artifact in
...
+The `Images` data model represents the information of an image which acts as a template for containers. The Observer sends this information to the Cloud Event Handler to store newly generate image. See [cloud events](cloudevents.hbs.md).
+
+Each `Image` data entry stores information about the associated location in which it was deployed, details about the state, and what image was used. See the diagram above for a list of all fields the data model stores.
+You can only associate an `Images` entry with one `Locations` entry. You can point multiple `Images` entries to the same `Locations` entry. 
+
+### <a id='commits'></a> Commits
+
+The `Commits` data model represents the information of a snapshot of the changes that have been made to a project's files. The Observer sends this information to the Cloud Event Handler to store new commits. See [cloud events](cloudevents.hbs.md).
+
+Each `Commit` data entry stores information about the associated location in which it was deployed, details about the state, and what image was used. See the diagram above for a list of all fields the data model stores.
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: Just doublechecking: `what image was used` - is this description accurate for commits?

File: scst-store/amr/data-model-and-concepts.hbs.md (hunk 7 above)
Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md]: Updated

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 9]
```diff
@@ -10,28 +10,58 @@ The following diagram shows the data models used in the AMR to store artifact in
...
+
+### <a id='appAcceleratorRuns'></a> AppAcceleratorRuns
+The `AppAcceleratorRuns` data model represents the information of new projects running from Git repositories. An accelerator.yaml file in the repository declares input options for the accelerator. This file also contains instructions for processing the files when you generate a new project. Observer sends this information to the Cloud Event Handler to store `AppAcceleratorRuns`. See [cloud events](cloudevents.hbs.md).
+
+### <a id='appAcceleratorFragments'></a> AppAcceleratorFragments
+The `AppAcceleratorFragments` Accelerator fragments are reusable accelerator components that can provide options, files, or transforms. They can be imported to accelerators using an `import` entry and the transforms from the fragment can be referenced in an `InvokeFragment` transform in the accelerator that is declaring the import. data model represents the information of a fragment of accelerator app. The Observer sends this information to the Cloud Event Handler to store `AppAcceleratorFragments`. See [cloud events](cloudevents.hbs.md).
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: `data model represents the information of a fragment of accelerator app.` - This is not quite a complete sentence

**Review by xtreme-debbie-chen — CHANGES_REQUESTED** [review]

File: scst-store/amr/data-model-and-concepts.hbs.md (hunk 9 above)
Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md]: Do you have any suggestions?

File: scst-store/amr/data-model-and-concepts.hbs.md (hunk 8 above)
Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md]: I think that was something suggested by @ohlookadollar

Comment (xtreme-debbie-chen) [reply scst-store/amr/data-model-and-concepts.hbs.md]: What is the relationship between the commit and "the image used"? I'm just not clear on what that means.

Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md]: Ah, I can make it more explicit `image` here is commit sha or tag.

File: scst-store/amr/data-model-and-concepts.hbs.md (hunk 3 above)
Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md]: Updated it

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 10]
```diff
@@ -10,28 +10,58 @@ The following diagram shows the data models used in the AMR to store artifact in
...
+
+### <a id='appAcceleratorFragments'></a> AppAcceleratorFragments
+The `AppAcceleratorFragments` Accelerator fragments are reusable accelerator components that can provide options, files, or transforms. They can be imported to accelerators using an `import` entry and the transforms from the fragment can be referenced in an `InvokeFragment` transform in the accelerator that is declaring the import. The AppAcceleratorFragments data model represents the information of a fragment in the accelerator app. The Observer sends this information to the Cloud Event Handler to store `AppAcceleratorFragments`. See [cloud events](cloudevents.hbs.md).
+
+`AppAcceleratorRuns` can have zero or many `AppAcceleratorFragments`. 
+
+### <a id='doraMetrics'></a> DoraMetrics
+The `DoraMetrics` data model represents the information of Dora Metric. Observer sends this information to the Cloud Event Handler to store `DoraMetrics`. See [cloud events](cloudevents.hbs.md).
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: I don't actually know but is `Dora` technically supposed to be `DORA`? I think it's an acronym, right?

Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: Also missed one! `Observer` should be `The Observer` based on the other examples in this doc.

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 11]
```diff
@@ -10,28 +10,58 @@ The following diagram shows the data models used in the AMR to store artifact in
...
+
+### <a id='doraMetricsPerCorrelationID'></a> DoraMetricsPerCorrelationID
+The `DoraMetricsPerCorrelationID` data model represents the information of Dora Metric for Correlation ID. The Observer sends this information to the Cloud Event Handler to store `DoraMetricsPerCorrelationID`. See [cloud events](cloudevents.hbs.md).
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md]: `Dora Metric for Correlation ID` -> `a DORA Metric for a specific Correlation ID`

Do we need to explain what correlation ID is? Or is that already defined elsewhere? If so, we should link to it.

Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md]: Explained it a little better, also it is explained in the diagram.

File: scst-store/amr/data-model-and-concepts.hbs.md [hunk 12]
```diff
@@ -10,28 +10,58 @@ The following diagram shows the data models used in the AMR to store artifact in
...
+
+`AppAcceleratorRuns` can have zero or many `AppAcceleratorFragments`. 
+
+### <a id='doraMetrics'></a> DoraMetrics
+The `DoraMetrics` data model represents the information of DORA Metric. The Observer sends this information to the Cloud Event Handler to store `DoraMetrics`. See [cloud events](cloudevents.hbs.md).
+
+### <a id='doraMetricsPerCorrelationID'></a> DoraMetricsPerCorrelationID
+The `DoraMetricsPerCorrelationID` data model represents the information of DORA Metric for Correlation ID. The Correlation ID that groups the all the artifacts together. The Observer sends this information to the Cloud Event Handler to store `DoraMetricsPerCorrelationID`. See [cloud events](cloudevents.hbs.md).
```
Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md:66]: `groups the all the artifacts` doesn't make sense

Comment (xtreme-debbie-chen) [inline scst-store/amr/data-model-and-concepts.hbs.md:66]: Do we need to explain what correlation ID is? Or is that already defined elsewhere that this page can link to?

Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md:66]: Based on a discussion with Tony, I have deleted the description for now. All these fields will be defined when creating the API documents in the future just like MDS

Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md:66]: Based on a discussion with Tony, I have deleted the description for now. All these fields will be defined when creating the API documents in the future just like MDS

**Review by Manifaust — APPROVED** [review]

File: scst-store/amr/data-model-and-concepts.hbs.md (hunk 12 above)
Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md:66]: As per discussion with @Manifaust we might more changes in the future but for now changes are good.

Comment (shaileshpandey) [reply scst-store/amr/data-model-and-concepts.hbs.md:66]: As per discussion with @Manifaust we might more changes in the future but for now changes are good.

**Pull request merged**